src/WebJobs.Script/Diagnostics/ConditionalTraceWriter.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using Microsoft.Azure.WebJobs.Host;
+
+namespace Microsoft.Azure.WebJobs.Script
+{
+    public sealed class ConditionalTraceWriter : TraceWriter
+    {
+        private readonly Func<TraceEvent, bool> _predicate;
+        private readonly FailedConditionTraceBehavior _failedConditionBehavior;
+
+        public ConditionalTraceWriter(TraceWriter innerWriter, Func<TraceEvent, bool> predicate, FailedConditionTraceBehavior failedConditionBehavior = FailedConditionTraceBehavior.TraceAsVerbose) 
+            : base(innerWriter?.Level ?? TraceLevel.Off)
+        {
+            if (innerWriter == null)
+            {
+                throw new ArgumentNullException(nameof(innerWriter));
+            }
+
+            if (predicate == null)
+            {
+                throw new ArgumentNullException(nameof(predicate));
+            }
+
+            InnerWriter = innerWriter;
+            _predicate = predicate;
+            _failedConditionBehavior = failedConditionBehavior;
+        }
+
+        public TraceWriter InnerWriter { get; }
+
+        public override void Trace(TraceEvent traceEvent)
+        {
+            if (_predicate(traceEvent))
+            {
+                InnerWriter.Trace(traceEvent);
+            }
+            else if (_failedConditionBehavior == FailedConditionTraceBehavior.TraceAsVerbose)
+            {
+                var eventClone = CloneEvent(traceEvent);
+
+                // To improve troubleshooting, instead of completely suppressing the event, we'll downgrade
+                // it to 'Verbose' and add a prefix to the message.
```

Comment:
Yes, that would be the current default behavior. I wanted to make sure we weren't still had the ability to capture those logs if needed. At least the messages will still be logged with the `[Supressed...` prefix.

I like the idea of having the default behavior be based on the debug flag. Perhaps that's just another enumeration value we add to `FailedConditionTraceBehavior` and make that the default once that flag is available.